cpp/src/arrow/compute/compute-test.cc

```diff
@@ -1056,6 +1056,37 @@ TEST_F(TestCast, StringToNumberErrors) {
   CheckFails<StringType, std::string>(utf8(), {"z"}, is_valid, float32(), options);
 }
 
+TEST_F(TestCast, StringToTimestamp) {
+  CastOptions options;
+
+  vector<bool> is_valid = {true, false, true, true};
```

Comment:
Does not match `strings` size. Worth adding a DCHECK in ArrayFromVector.

Reply:
Hmm, good catch.